install/infra/modules/k3s/dns.tf

```diff
@@ -0,0 +1,21 @@
+resource "google_dns_managed_zone" "gitpod-dns-zone" {
+  count         = var.domain_name == null ? 0 : 1
+  name          = "zone-${var.name}"
+  dns_name      = "${var.domain_name}."
+  description   = "Terraform managed DNS zone for ${var.name}"
+  force_destroy = true
+  labels = {
+    app = "gitpod"
+  }
+}
+
+resource "google_dns_record_set" "gitpod-dns" {
+  for_each     = local.dns_list
+  name         = each.key
+  managed_zone = google_dns_managed_zone.gitpod-dns-zone[0].name
```

Comment:
**question:** Do we anticipate the need to use a pre-existing Cloud DNS zone? The release preview environments (eg `release-<xxx>.tests.gitpod-self-hosted.com`) domains use a pre-existing domain, will this code work when we stand up preview environments where the Cloud DNS zone is managed outside of this module?

Reply:
This is not a use case we anticipate. Currently in GCP, AWS and Azure the behavior is that we create a managed zone by the name and add the NS record. I am only making the behavior consistent here.